htlcswitch/switch.go

```diff
@@ -300,11 +309,39 @@ func (s *Switch) handleLocalDispatch(payment *pendingPayment, packet *htlcPacket
 	case *lnwire.UpdateFailHTLC:
 		// Retrieving the fail code from byte representation of error.
 		var userErr error
-		if code, err := htlc.Reason.ToFailCode(); err != nil {
-			userErr = errors.Errorf("can't decode fail code id"+
-				"(%v): %v", htlc.ID, err)
+
+		failure, err := payment.deobfuscator.Deobfuscate(htlc.Reason)
+		if err != nil {
+			userErr = errors.Errorf("unable to de-obfuscate "+
+				"onion failure, htlc with id(%v): %v", htlc.ID, err)
```

Comment:
I'd say the payment hash of the HTLC is more relevant to a user than the ID (which they don't control or set).